numpyro/distributions/discrete.py

```diff
@@ -183,12 +183,14 @@ def support(self):
         return constraints.integer_interval(0, self.total_count)
 
     def enumerate_support(self, expand=True):
-        total_count = jnp.amax(self.total_count)
-        if not_jax_tracer(total_count):
+        if not_jax_tracer(self.total_count):
+            total_count = np.amax(self.total_count)
```

Comment:
Same as my comment below - is this `np` needed because `self.total_count` could be a python scalar. In that case does it make sense to use `jnp.array` or `device_put` everywhere? That way, we can freely use `jnp` functions instead of a mix of `np` and `jnp` which becomes hard to reason about.